doc/xml/parsing.md

```diff
@@ -0,0 +1,504 @@
+
+```
+xml = <<DOCTYPE
+<!DOCTYPE note [
+<!ELEMENT note (to,from,heading,body)>
+<!ELEMENT to (#PCDATA)>
+<!ELEMENT from (#PCDATA)>
+<!ELEMENT heading (#PCDATA)>
+<!ELEMENT body (#PCDATA)>
+]>
+DOCTYPE
+doc = Nokogiri::XML.parse(xml)
+doc
+# =>
+#(Document: {
+  name = "document",
+  children = [
+    #(DTD: {
+      name = "note",
+      children = [
+        #(ElementDecl: { "<!ELEMENT note (to , from , heading , body)>\n" }),
+        #(ElementDecl: { "<!ELEMENT to (#PCDATA)>\n" }),
+        #(ElementDecl: { "<!ELEMENT from (#PCDATA)>\n" }),
+        #(ElementDecl: { "<!ELEMENT heading (#PCDATA)>\n" }),
+        #(ElementDecl: { "<!ELEMENT body (#PCDATA)>\n" })]
+      })]
+  })
+  ```
+
+### Attribute-List Declarations
+
+\Nokogiri parses an [XML attribute-list declaration](https://www.w3.org/TR/REC-xml/#attdecls)
+into a Nokogiri::XML::AttributeDecl object:
+
+```
+xml = <<DOCTYPE
+<!DOCTYPE note [
+<!ELEMENT payment (#PCDATA)>
+<!ATTLIST payment type CDATA "check">
+]>
+DOCTYPE
+doc = Nokogiri::XML.parse(xml)
+doc
+# =>
+#(Document: {
+  name = "document",
+  children = [
+    #(DTD: {
+      name = "note",
+      children = [
+        #(ElementDecl: { "<!ELEMENT payment (#PCDATA)>\n" }),
+        #(AttributeDecl: { "<!ATTLIST payment type CDATA \"check\">\n" })]
+      })]
+  })
+```
+
+### Conditional Sections
+
+\Nokogiri parses an [XML conditional section](https://www.w3.org/TR/REC-xml/#sec-condition-sect)
+into a Nokogiri::XML::EntityDecl object:
+
+```
+xml = <<DOCTYPE
+<!DOCTYPE note [
+<!ENTITY % draft 'INCLUDE' >
+]>
+DOCTYPE
+doc = Nokogiri::XML.parse(xml)
+doc
+# =>
+#(Document: {
+  name = "document",
+  children = [
+    #(DTD: {
+      name = "note",
+      children = [ #(EntityDecl: { "<!ENTITY % draft \"INCLUDE\">\n" })]
+      })]
+  })
+```
+
+### Character References
+
+\Nokogiri parses an [XML character reference](https://www.w3.org/TR/REC-xml/#sec-references)
+(such as <tt>&#9792;</tt>)
+and replaces it with a character such as (<tt>'♀'</tt>):
+
+```
+xml = <<ELE
+<root>
+  <name>
+    <vorname>Marie</vorname>
+    <nachname>M&#252;ller</nachname>
+    <geschlecht>&#9792;</geschlecht>
+  </name>
+</root>
+ELE
+doc = Nokogiri::XML.parse(xml)
+doc
+# =>
+#(Document: {
+  name = "document",
+  children = [
+    #(Element: {
+      name = "root",
+      children = [
+        #(Text "\n  "),
+        #(Element: {
+          name = "name",
+          children = [
+            #(Text "\n    "),
+            #(Element: { name = "vorname", children = [ #(Text "Marie")] }),
+            #(Text "\n    "),
+            #(Element: { name = "nachname", children = [ #(Text "Müller")] }),
+            #(Text "\n    "),
+            #(Element: { name = "geschlecht", children = [ #(Text "♀")] }),
+            #(Text "\n  ")]
+          }),
+        #(Text "\n")]
+      })]
+  })
+```
+
+### Entity References
+
+\Nokogiri parses an [XML entity reference](https://www.w3.org/TR/REC-xml/#sec-references)
+(such as <tt>&lt;</tt>)
```

Comment:
Same note as above re: escaping for web browsers

Reply:
Fixed.